File: trabajos_previos/sesion_3/recursividad.cpp
#include <iostream>
using namespace std;

int factorial(int);

int main () {
    // Esta parte solo es para declarar e inicializar las variables
    // además de mostrar el resultado
    int n, result;

    cout << "Enter a non-negative number: ";
    cin>> n;
    
    result = factorial(n);
    cout << "Factorial of " << n << " = " << result;
    return 0;
}
// Función del factorial usando recursividad
int factorial(int n){
    if (n > 1){
        // Se llama la función recursivamente en (n-1) multiplicada por n mientras n > 1
        return n * factorial(n-1);
    }else{
        return 1;
    }
}

File: trabajos_previos/semana_4/clase_11/recursividad.py
# Recursividad: esta es la sintaxis
def factorial(x):
    '''Para encontrar el factorial de un número'''
    if x == 1:
        return 1
    else:
        # x por el factorial del anterior
        return (x * factorial(x-1))
    
# Llama con 3
num = 3
print("The factorial of", num, "is", factorial(num))
